cql3/result_set.hh

```diff
@@ -193,6 +194,27 @@ public:
         }
     }
 
+    // visit_gently() is like visit(), except it may yield between rows so
+    // returns a future that will resolve when it's done. We only yield
+    // between rows, not between individual cells, which is a good compromise
+    // if we assume that individual rows are not too large.
+    template<typename Visitor>
+    requires ResultVisitor<Visitor>
+    future<> visit_gently(Visitor& visitor) const {
```

Comment:
You can write: `future<> visit_gently(ResultVisitor auto &visitor) const {`, removing two lines and a use-once name.

Reply:
Right, will do this. I just copied the signature of the existing visit() function, without thinking about it too much.